frontend/src/routes/Search/SearchResults/utils.tsx

```diff
@@ -27,6 +34,42 @@ export interface ISearchResult {
   __type: string
 }
 
+export function handleResourceActionAuth(
+  action: 'create' | 'update' | 'delete',
+  kind: string,
+  cluster: string,
+  apiVersion: string,
+  name?: string,
+  namespace?: string
+) {
+  switch (kind.toLowerCase()) {
+    case 'virtualmachine':
+      return canUser(
+        'create',
+        {
+          // Map VM action authentication to MCA create resource.
+          apiVersion: ManagedClusterActionApiVersion,
+          kind: ManagedClusterActionKind,
+        },
+        cluster === 'local-cluster' ? namespace : cluster
+      )
+    default:
+      return canUser(
+        action,
+        {
+          apiVersion,
+          kind,
+          metadata: {
+            name,
+            namespace,
+          },
+        },
+        cluster === 'local-cluster' ? namespace : cluster,
+        name
+      )
+  }
+}
```

Comment:
Shouldn't this just be based on whether the resource is on the local-cluster or a managed cluster?

Reply:
@jlpadilla Correct me if i'm wrong here, but we don't want to provide explicit auth on VM/VMI resources. To prevent users from creating resources on hub or elsewhere they shouldn't be able to?

Reply:
I believe that @KevinFCormier point is that the ACM hub cluster could also have the Openshift Virtualization operator. So there's 2 scenarios:
- **VM/VMI object is in the ACM Hub cluster.** 
  For this scenario we don't use the ManagedServiceAccount. For RBAC we check directly with the user account. `oc auth can-i put virtualmachine/start -n vmnamespace`
- **VM/VMI object is in a Managed cluster.**
  For this scenario we use the ManagedServiceAccount and map RBAC to ManagedClusterAction.

The logic will be akin to what we do for other resources like Pod.

Reply:
My point was that I thought this code was used for all resources from search results, not just VirtualMachines. So for any resource that's not on the local-cluster, wouldn't we be taking the action via ManagedClusterAction?

But I have just skimmed, so I could be wrong here.

Reply:
@zlayne I see where's the confusion. **Edit & Delete** should work the same as all other resources. I don't expect any changes for these 2 actions.

The changes should be for the new actions [start, stop, restart, pause, unpause].

Reply:
@jlpadilla Oh my bad on the confusion here. For the new VM actions I handle auth in the backend [here](https://github.com/stolostron/console/blob/main/backend/src/routes/virtualMachineProxy.ts#L42). I am not currently disabling the action item in the UI, as the UI only authenticates in user token.